Sources/SwiftProtobuf/ProtoNameProviding.swift

```diff
@@ -36,6 +36,14 @@ public protocol ProtoNameProviding {
   /// instance method allows generated messages to override it and ask their
   /// extension sets for names as well.
   func _protobuf_fieldNumber(forProtoName name: String) -> Int?
+
+  /// Returns the field number for the field with the given json name.
+  ///
+  /// The default implementation looks up the field in the static name map,
+  /// which is sufficient for proto3. For proto2 extensions, making this an
```

Comment:
Good point.

This implies that there should not be any reason to override this lookup.